pyk/src/pyk/cfg_manager.py

```diff
@@ -0,0 +1,247 @@
+            for sort, arg in zip(production_arity, _kast.args):
+                if sort.name.endswith('Cell') and type(arg) is KVariable:
+                    new_args.append(definition.empty_config(sort))
+                else:
+                    new_args.append(arg)
+            return KApply(_kast.label, new_args)
+        return _kast
+    return bottom_up(_cellVarsToLabels, term)
+
+
+def rename_generated_vars(cterm: CTerm) -> CTerm:
+    state, *constraints = cterm
+    _, config_subst = splitConfigFrom(state)
+    config_var_count = {cvar: count_vars(ccontents) for cvar, ccontents in config_subst.items()}
+    free_vars = collectFreeVars(cterm.term)
+    var_subst: Dict[str, KInner] = {}
+    for v in free_vars:
+        if v.startswith('_Gen') or v.startswith('?_Gen') or v.startswith('_DotVar') or v.startswith('?_DotVar'):
+            cvars = [cv for cv in config_var_count if v in config_var_count[cv]]
+            if len(cvars) > 1:
+                raise ValueError(f'Found "Gen*" or "DotVar*" variable with multiple occurances: {v}')
+            cvar = cvars[0]
+            new_v = abstract_term_safely(KVariable(v), base_name=cvar)
+            while new_v.name in free_vars:
+                new_v = abstract_term_safely(KVariable(new_v.name), base_name=cvar)
+            var_subst[v] = new_v
+            free_vars.append(new_v.name)
+    return CTerm(Subst(var_subst).apply(cterm.term))
+
+
+def sanitize_config(defn: KDefinition, init_term: KInner) -> KInner:
+
+    def _var_name(vname):
+        new_vname = vname
+        while new_vname.startswith('_') or new_vname.startswith('?'):
+            new_vname = new_vname[1:]
+        return new_vname
+
+    free_vars_subst = {vname: KVariable(_var_name(vname)) for vname in collectFreeVars(init_term)}
+
+    # TODO: This is somewhat hacky. We shouldn't have to touch the config this much.
+    # Likely, the frontend should just be giving us initial states with these already in place.
+    def _remove_cell_map_definedness(_kast):
+        if type(_kast) is KApply:
+            if _kast.label.name.endswith('CellMap:in_keys'):
+                return Bool.false
+            elif _kast.label.name.endswith('CellMapItem'):
+                return _kast.args[1]
+        return _kast
+
+    new_term = substitute(init_term, free_vars_subst)
+    new_term = remove_generated_cells(new_term)
+    new_term = bottom_up(_remove_cell_map_definedness, new_term)
+
+    if new_term not in [mlTop(), mlBottom()]:
+        config, constraint = split_config_and_constraints(new_term)
+        constraints = [bool_to_ml_pred(ml_pred_to_bool(c, unsafe=True)) for c in flatten_label('#And', constraint)]
+        new_term = mlAnd([config] + constraints)
+
+    return new_term
+
+
+class CFGManager:
+    summary_name: Final[str]
+    strategy_name: Final[str]
+    spec_file: Final[Path]
+    main_file: Final[Optional[Path]]
+    summary_dir: Final[Path]
+    config_file: Final[Path]
+    kompiled_dir: Final[Path]
```

Comment:
We're using `definition_dir` now instead of `kompiled_dir` for consistency with the new `--definition` flag to the frontend.